### PR TITLE
Changed for each to for iterator

### PR DESCRIPTION
**Pull request by randomuserJ**

### CODE REVIEW

File: circleGroup.cpp [hunk 1]
```diff
@@ -47,20 +47,24 @@ void CircleGroup::generatePoints()
 
 const CirclePoint& CircleGroup::getInverse(const CirclePoint& pointToFindInv) const
 {
+    std::cout << *this;
+
```
Comment (RandomUserIK) [inline circleGroup.cpp]: Remove this.

File: circleGroup.cpp [hunk 2]
```diff
@@ -47,20 +47,24 @@ void CircleGroup::generatePoints()
...
         return this->points.at(0);
 
     if (pointToFindInv.getOrd() == 2)
         return this->points.at(1);
 
-    for (size_t i = 1; i < this->points.size(); i++)
+    for(auto iterator = this->points.begin(); iterator != this->points.end(); iterator++)
     {
-        if (pointToFindInv == this->points.at(i))
+        if (pointToFindInv == (*iterator))
```
Comment (RandomUserIK) [inline circleGroup.cpp]: No need to use brackets (*iterator) == *iterator.

Comment (RandomUserIK) [inline circleGroup.cpp]: No need to use brackets (*iterator) == *iterator.

Comment (RandomUserIK) [inline circleGroup.cpp]: No need to use brackets (*iterator) == *iterator.

File: circleGroup.cpp [hunk 3]
```diff
@@ -47,20 +47,24 @@ void CircleGroup::generatePoints()
 
 const CirclePoint& CircleGroup::getInverse(const CirclePoint& pointToFindInv) const
 {
+    std::cout << *this;
```
Comment (RandomUserIK) [inline circleGroup.cpp]:
```suggestion
    // std::cout << *this;
```
Remove this

Comment (RandomUserIK) [inline circleGroup.cpp]:
```suggestion
    std::cout << *this;
```
Remove this.

**Comment by Pezeter** [comment]
Did some refactoring of getInverse method, please review the changes.

**Pull request merged**